go/vt/vttablet/tabletmanager/rpc_actions.go

```diff
@@ -76,28 +75,27 @@ func (agent *ActionAgent) changeTypeLocked(ctx context.Context, tabletType topod
 		return fmt.Errorf("Tablet: %v, is already drained", agent.TabletAlias)
 	}
 
-	agentMasterTermStartTime := time.Time{}
+	tablet := agent.Tablet()
+	tablet.Type = tabletType
 	// If we have been told we're master, set master term start time to Now
+	// and save it topo immediately.
 	if tabletType == topodatapb.TabletType_MASTER {
-		agentMasterTermStartTime = time.Now()
-	}
-	// change our type in the topology, and set masterTermStartTime on tablet record if applicable
-	_, err := topotools.ChangeType(ctx, agent.TopoServer, agent.TabletAlias, tabletType, logutil.TimeToProto(agentMasterTermStartTime))
-	if err != nil {
-		return err
-	}
-	// We only update agent's masterTermStartTime if we were able to update the topo.
-	// This ensures that in case of a failure, we are never in a situation where the
-	// tablet's timestamp is ahead of the topo's timestamp.
-	if tabletType == topodatapb.TabletType_MASTER {
+		agentMasterTermStartTime := time.Now()
+		tablet.MasterTermStartTime = logutil.TimeToProto(agentMasterTermStartTime)
+
+		// change our type in the topology, and set masterTermStartTime on tablet record if applicable
+		_, err := topotools.ChangeType(ctx, agent.TopoServer, agent.TabletAlias, tabletType, tablet.MasterTermStartTime)
+		if err != nil {
+			return err
+		}
+		// We only update agent's masterTermStartTime if we were able to update the topo.
+		// This ensures that in case of a failure, we are never in a situation where the
+		// tablet's timestamp is ahead of the topo's timestamp.
 		agent.setMasterTermStartTime(agentMasterTermStartTime)
 	}
 
```

Comment:
We need an else part for non-master types (to set the MasterTermStartTime to 0/nil).